src/FlaUI.Core/AutomationElements/TextBox.cs

```diff
@@ -48,10 +73,70 @@ public string Text
                 }
                 else
                 {
-                    Enter(value);
+                    if (SetTextWin32(value) == false)
+                    {
+                        Enter(value);
+                    }
                 }
             }
         }
+        
+        private bool SetTextWin32(string text)
+        {
+		// try with native Win32 function SetWindowText
+		if (Properties.NativeWindowHandle.IsSupported)
+		{
+			var windowHandle = Properties.NativeWindowHandle.ValueOrDefault;
+			if (windowHandle != IntPtr.Zero)
+			{
+				IntPtr textPtr = IntPtr.Zero;
+				try
+				{
+					textPtr = Marshal.StringToBSTR(text);
+				}
+				catch { }
+
+				try
+				{
+					if (textPtr != IntPtr.Zero)
+					{
+						if (SendMessage(windowHandle, WindowMessages.WM_SETTEXT, IntPtr.Zero, textPtr) ==
+							Win32Constants.TRUE)
+						{
+							return true; // text successfully set
+						}
+					}
+				}
+				catch { }
+				finally
+				{
+					Marshal.FreeBSTR(textPtr);
+				}
+			}
+		}
+		return false;
+        }
+	
+	private string GetTextWin32()
+	{
+		// try with native Win32 function SetWindowText
+		if (Properties.NativeWindowHandle.IsSupported)
+		{
+			var windowHandle = Properties.NativeWindowHandle.ValueOrDefault;
+			if (windowHandle != IntPtr.Zero)
+			{
+				IntPtr textLengthPtr = SendMessage(hwnd, WindowMessages.WM_GETTEXTLENGTH, IntPtr.Zero, IntPtr.Zero);
+				if (textLengthPtr.ToInt32() > 0)
+				{
+					int textLength = textLengthPtr.ToInt32() + 1;
+					StringBuilder text = new StringBuilder(textLength);
+					SendMessage(hwnd, WindowMessages.WM_GETTEXT, textLength, text);
+					return text.ToString();
+				}
+			}
+		}
+		return null;
```

Comment:
Can't it be null in case there is no text? Is it String.Empty in that case? otherwise you should return a boolean indicating success / failure.

Reply:
I changed and return boolean.